### ISSUE TITLE
Display CO₂ estimate per million tokens

### ISSUE THREAD

**Issue by KennethEnevoldsen** [embeddings-benchmark/mteb#4368]
### Description of the feature

We have active parameters for most models in MTEB, which is sufficient to estimate inference cost as gCO₂ / Mtoken. This is a useful signal for model selection that no other embedding leaderboard currently surfaces.
Proposed display

A compact badge (e.g. ~42 gCO₂ / Mtoken) in the model documentation page. On hover, a popover expands showing the assumptions behind the estimate:
- Active parameters
- Benchmark hardware (e.g. A100 80GB)
- Carbon intensity (400 gCO₂/kWh, ADEME world 
- PUE (1.2)

<img width="764" height="286" alt="Image" src="https://github.com/user-attachments/assets/4756d84f-b543-453c-9fef-efea1edb05ee" />

All estimates are prefixed with ~ — we do not have visibility into deployment hardware or grid carbon intensity, so confidence should not be overstated for any model. The footnote in the popover makes clear that the figure reflects benchmark conditions, not real-world deployment.

For closed models where active parameters are not disclosed, we have two options: (1) fall back to EcoLogits estimates, or (2) estimate active parameters from observed token throughput and known hardware specs, since inference latency is roughly linear in active parameters. Either way, the assumption should be noted explicitly in the popover.

**Methodology**

To be defined, but we could follow the EcoLogits linear regression structure (E_per_token = a * P_active + b) fitted on embedding-specific benchmark data, so that our estimates are on the same scale as EcoLogits figures for closed generative models. 

**References**

- EcoLogits methodology: https://ecologits.ai/0.2/methodology/llm_inference/
- Luccioni et al. (2024) — “Power Hungry Processing: Watts Driving the Cost of AI Deployment?” — the most directly relevant prior work; measures deployment cost as energy and carbon required to perform 1,000 inferences across task-specific and general-purpose models, finding that generative architectures are orders of magnitude more expensive than task-specific systems : https://arxiv.org/abs/2311.16863
- Faiz et al. (2024) — “LLMCarbon: Modeling the end-to-end Carbon Footprint of Large Language Models” — covers both dense and MoE LLMs including embodied carbon: https://arxiv.org/abs/2309.14393
- Luccioni et al. (2022) — “Estimating the Carbon Footprint of BLOOM” — foundational methodology for inference carbon accounting including PUE: https://jmlr.org/papers/volume24/23-0069/23-0069.pdf

Related to #4366

**Comment by Samoed** [comment]
Isn't this metric is very biased due to different gpu architectures and different optimizations?

**Comment by KennethEnevoldsen** [comment]
Yeah I would provide that in the "benchmark hardware", e.g. A100 to indicate what we measure based on and Luccioni et al. does use an estimate based on active parameters (which is not perfect). Before I implemented anywhere, I would probably see how well it aligned with metrics from CodeCarbon.

edit: but in general I think it is more about the relative scale

**Comment by KennethEnevoldsen** [comment]
from comment https://github.com/embeddings-benchmark/mteb/pull/4918#issuecomment-5135047442

---

So had quite a long discussion on this here:

https://github.com/betagouv/ComparIA/issues/505

the short form: CO2 is very context dependent, what we really care about (and can measure) is energy efficiency, we can do this by splitting the models into energy categories (in the US I believe this use stars instead of letters). Here is the example for **generative** models:

| Class | Wh / 1M tokens |
| ----- | -------------- |
| A     | `< 100`        |
| B     | `100 - 300`    |
| C     | `300 - 1 000`  |
| D     | `1 000 - 3 000`|
| E     | `3 000 - 10 000`|
| F     | `>= 10 000`    |

Which gives something like this:

| Model shape           | Energy     | Class |
| --------------------- | ---------- | ----- |
| 8B dense              | ~90 Wh     | A     |
| 24-35B dense          | ~110-250   | B     |
| 70B dense             | ~660 Wh    | C     |
| 120B MoE (5B active)  | ~340 Wh    | C     |
| 400B MoE (17B active) | ~1 600 Wh  | D     |
| 700B dense            | ~30 000 Wh | F     |

For more discussion see the issue, but generally I think we should focus on energy usage and then create our energy class bands. One thing we might need to consider which they do not is embedding size (which for us has downstream effect)

Let me know what you think about this alternative. For me it is easier to justify than co2.
